common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationStrategy.java

```diff
@@ -50,6 +58,46 @@
 
     public static final String REQUEST_CODE = "com.microsoft.identity.client.request.code";
 
+    private WeakReference<Context> mReferencedApplicationContext;
+    private WeakReference<Activity> mReferencedActivity;
+    private WeakReference<Fragment> mReferencedFragment;
+
+    /**
+     * Constructor of AuthorizationStrategy.
+     */
+    public AuthorizationStrategy(@NonNull Context applicationContext,
+                                 @NonNull Activity activity,
+                                 @Nullable Fragment fragment) {
+        mReferencedApplicationContext = new WeakReference<>(applicationContext);
+        mReferencedActivity = new WeakReference<>(activity);
+        mReferencedFragment = new WeakReference<>(fragment);
+    }
+
+    protected Context getApplicationContext() {
+        return mReferencedApplicationContext.get();
+    }
+
+    /**
+     * If fragment is provided, add AuthorizationFragment on top of that fragment.
+     * Otherwise, launch AuthorizationActivity.
+     */
+    protected void launchIntent(@NonNull Intent intent) {
```

Comment:
This method is used both by `EmbeddedWebViewStrategy` and `BrowserStrategy`. Shouldn't this be only for `Webview` ? i.e. in case of browser, what does adding a fragment do ?

Reply:
AuthorizationFragment will be launched, which in turn would launch CustomTabsActivity - inside onResume().

CustomTabsActivity will be popped up. It won't be displayed in-line in the specified fragment area.